### PR TITLE
Feature/p2022 628 mock repo with list of products

### PR DESCRIPTION
**Pull request by pawelsanaluta**
# P2022-628

## PROBLEM
Main goal of this task was to provide user with content using mocked data model.

## SOLUTION
To achieve that a json file was created contatining list of 10 exemplary products. Json file is being converted to list of ProductAPI by function getProductAPIList() in JsonFetcher class. Result of this function is mapped using map function in ProductListMapper class to a list of Product. Logic of that process is placed in ProductRepositoryImpl class which implements intarface ProductRepository that contains method getProductList().

### CODE REVIEW

File: app/src/main/java/com/intive/patronage22/lublin/repository/ProductRepository.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.intive.patronage22.lublin.repository
+
+import com.intive.patronage22.lublin.repository.model.Product
+
+interface ProductRepository {
+    fun getProductList(): List<Product>
```
Comment (dziubinski-intive) [inline app/src/main/java/com/intive/patronage22/lublin/repository/ProductRepository.kt:6]: repository should pretend real network call, please make it suspend function

File: app/src/main/java/com/intive/patronage22/lublin/repository/mapper/Mapper.kt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package com.intive.patronage22.lublin.repository.mapper
+
+interface Mapper<I, O> {
```
Comment (dziubinski-intive) [inline app/src/main/java/com/intive/patronage22/lublin/repository/mapper/Mapper.kt:3]: this abstraction does not introduce any value here, if we are not using this interface limitation. It just forces us to pass single object, new object will be returned.

Personally would remove it.

File: app/src/main/java/com/intive/patronage22/lublin/repository/ProductRepositoryImpl.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.intive.patronage22.lublin.repository
+
+import android.content.Context
+import com.intive.patronage22.lublin.repository.mapper.ProductListMapper
+import com.intive.patronage22.lublin.repository.model.Product
+
+class ProductRepositoryImpl(var context: Context) : ProductRepository {
```
Comment (dziubinski-intive) [inline app/src/main/java/com/intive/patronage22/lublin/repository/ProductRepositoryImpl.kt:7]: please rename to `ProductRepositoryMock`

File: app/src/main/java/com/intive/patronage22/lublin/repository/ProductRepository.kt (hunk 1 above)
Comment (dziubinski-intive) [reply app/src/main/java/com/intive/patronage22/lublin/repository/ProductRepository.kt:6]: that will help us to integrate Retrofit later

Comment (pawelsanaluta) [reply app/src/main/java/com/intive/patronage22/lublin/repository/ProductRepository.kt]: suspend keyword is added

File: app/src/main/java/com/intive/patronage22/lublin/repository/mapper/Mapper.kt (hunk 2 above)
Comment (pawelsanaluta) [reply app/src/main/java/com/intive/patronage22/lublin/repository/mapper/Mapper.kt]: function map() in Mapper interface was modified to take parameter List<ProductAPI> and the funtion's return type was changed to List<Product>

**Review by dziubinski-intive — APPROVED** [review]

**Pull request merged**